src/tasks/LibraryBuilder/Templates/preloaded_assemblies.c

```diff
@@ -0,0 +1,30 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+#include <stdlib.h>
+
+#include <mono/metadata/assembly.h>
+
+#ifdef HOST_ANDROID
+#include <android/log.h>
+
+#define LOG_ERROR(fmt, ...) __android_log_print(ANDROID_LOG_ERROR, "MONO_SELF_CONTAINED_LIBRARY", fmt, ##__VA_ARGS__)
+#else
+#include <os/log.h>
+
+#define LOG_ERROR(fmt, ...) os_log_error (OS_LOG_DEFAULT, fmt, ##__VA_ARGS__)
+#endif
+
+void
+mono_assembly_load_with_partial_name_check (const char* filename)
```

Comment:
this is only internal for this source file, correct? If so make it static and drop mono prefix.